### PR TITLE
THREESCALE-10749

### PR DESCRIPTION
**Pull request by austincunningham**
# What
Create a product with user key auth

# Issue
[THREESCALE-10749 ](https://issues.redhat.com/browse/THREESCALE-10749)

# Verification

- Setup 3scale
```bash
./setup_db.sh
./setup_apimanager.sh
```
- run the setup_product.sh
```bash
chmod 771 setup_product.sh
./setup_product.sh
```
- Confirm in the ui that the product is up and has a user key

image

- Test the `/nothing/<random_number>` endpoint for post and get requests e.g.
```bash
curl -X POST "https://product1-3scale-apicast-staging.apps.aucunnin.3bjq.s1.devshift.org/nothing/2030923?user_key=c00cc360c9896fe85e93a482fa724d53"
curl -X GET "https://product1-3scale-apicast-staging.apps.aucunnin.3bjq.s1.devshift.org/nothing/2030923?user_key=c00cc360c9896fe85e93a482fa724d53"
```

### CODE REVIEW

File: setup_product.sh [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  productCRName: product1-cr
+  production: true
+  deleteCR: true
+EOF
+
+echo Product Route: 
+echo "https://$(oc get routes | grep product1 |grep production| awk '{print $2}')" 
+echo
+echo User_key: 
+echo $(curl -s -X 'GET' "https://3scale-admin.$DOMAIN/admin/api/applications.xml?access_token=$ADMIN_ACCESS_TOKEN&page=1&per_page=500&service_id=3" -H 'accept: */*' | grep -oP '<user_key>\K[^<]+' | sed 's/\s//g')
```
Comment (carlkyrillos) [inline setup_product.sh:160]: This command didn't print the user key when I ran it the first time but the command itself works. I think we need to add a small sleep command at some point before it.

File: setup_product.sh [hunk 2]
```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
+
+# Get the ADMIN_URL and ADMIN_ACCESS_TOKEN from apimanger and system-seed secret
```
Comment (carlkyrillos) [inline setup_product.sh:3]:
```suggestion
# Get the ADMIN_URL and ADMIN_ACCESS_TOKEN from apimanager and system-seed secret
```
Small typo

**Review by carlkyrillos — COMMENTED** [review]
Added two small comments but the script worked as expected

File: setup_product.sh (hunk 1 above)
Comment (austincunningham) [reply setup_product.sh:161]: hmm wierd issue where after I put a sleep in it returns 2 keys . :exploding_head:

Comment (austincunningham) [reply setup_product.sh:161]: Not something I can fix here.

Comment (carlkyrillos) [reply setup_product.sh:161]: Yep looks like an operator issue. I think just open a ticket and ignore it on here.

**Comment by carlkyrillos** [comment]
Reviewed the latest changes and confirmed the sleep fixes the user_key not printing issue
/lgtm

File: setup_product.sh (hunk 1 above)
Comment (austincunningham) [reply setup_product.sh:161]: https://issues.redhat.com/browse/THREESCALE-10770

**Pull request merged**